### PR TITLE
Improve evolution.rst

### PR DESCRIPTION
**Pull request by mppf**
* add summary section
* adjust other examples for => deprecation
* call it "default intent" rather than "blank intent"
* include section on => deprecation
* make sure to describe old and new behavior
* other minor wording / spacing changes

Reviewed by @bradcray and @benharsh - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@bradcray - what do you think?

File: doc/rst/language/evolution.rst [hunk 1]
```diff
@@ -27,28 +52,72 @@ their scope is now a user error:
...
 .. code-block:: chapel
 
   proc badBeginSlice() {
     var A: [1..10000] int;
-    var slice => A[1..1000];
+    // slice1 sets up a slice using the => operator
+    // note that the => operator is deprecated (see below)
+    var slice1 => A[1..1000];
+    // slice2 sets up a slice by creating a reference to it
+    ref slice2 A[1..1000];
```
Comment (bradcray) [inline doc/rst/language/evolution.rst]: I think you're missing an equals here?

File: doc/rst/language/evolution.rst [hunk 2]
```diff
@@ -27,28 +52,72 @@ their scope is now a user error:
...
+For example, before Chapel 1.15 you might have written:
+
+.. code-block:: chapel
+
+  // pre-1.15
+  var A:[1..10] int;
+  // set up a const alias of A
+  const alias => A;
+  // set up a mutable slice of A
+  var slice => A[2..5];
```
Comment (bradcray) [inline doc/rst/language/evolution.rst]: You should probably add a third example of using the reindexing-based '=>' form:

````chapel
var A0: [0..9] = A;
````

which would now be written:

````chapel
ref A0 = A.reindex({0..9});
````

**Comment by bradcray** [comment]
Thanks for tackling this.  Someone should give it a closer read than I did.  I just scanned it quickly and happened to notice these few things.

File: doc/rst/language/evolution.rst (hunk 2 above)
Comment (mppf) [inline doc/rst/language/evolution.rst]: Did you mean
``` chapel
var A0: [0..9] => A;
```

?

Comment (bradcray) [inline doc/rst/language/evolution.rst]: Yes, sorry!

**Comment by mppf** [comment]
maybe @benharsh can look at it?

File: doc/rst/language/evolution.rst [hunk 3]
```diff
@@ -27,28 +52,77 @@ their scope is now a user error:
...
   proc badBeginSlice() {
     var A: [1..10000] int;
-    var slice => A[1..1000];
+    // slice1 sets up a slice using the => operator
+    // note that the => operator is deprecated (see below)
+    var slice1 => A[1..1000];
+    // slice2 sets up a slice by creating a reference to it
+    ref slice2 = A[1..1000];
+    // either way, using the slice in a begin that can continue
+    // after the function declaring theh array exits is an error
```
Comment (benharsh) [inline doc/rst/language/evolution.rst]: Typo: ``theh``

**Pull request merged**